### DISCUSSION TITLE
LED layout not working (Two large gaps)

### DISCUSSION THREAD

**Discussion by RSeyda** [awawa-dev/HyperHDR#907] [Q&A]
Hi!

This is my first time working with a raspberry pi 4 as I am a beginner at DIY projects in general so please bear with me.

I am using a ws2815, raspberry pi 4, a 3.3V to 5V level shifter, and a capture card. I am using the official raspberry pi power adapter and a separate power supply for the led strips (12V 10A). I have connected one of the pi's ground pins to the ground pin on the led strip so they share a common ground (It is in the - of the barrel jack). The BI is also unused.

I have done a couple of tests to confirm that everything is working as it is supposed to on the hardware side. When the system boots up, the swirl effect turns on and ALL of the LEDs are turned on and create a beautiful swirl effect. However, once the effect is over, the LEDs start acting weird.

I have set the correct amount of LEDs on each side (top, bottom, left and right), and set the correct input position. But for some reason the only leds that light up are most of the left and right-hand side strips and around 30% of the top and bottom strips. The rest of the LEDs lit up white until I turned on sparse processing which seemed to permanently turn the white "unused" LEDs off.

![ambilight problem](https://github.com/user-attachments/assets/b36a0fd2-9f98-4257-a0e9-86c06b931ed0)

My layout is: 77 top, 78 bottom, 42 left & right (total 239, maximum LED count is set to 256, nothing changed when I tried setting it to 239)

The LEDs that are lit up match the top and bottom rows. The right (*upside down*) "L" shape corresponds to the bottom row and the left corresponds to the top row. However, the right L-shape only has 70 lit up LEDs and the left L-shape only has 69 lit up. The gaps are about 38 LEDs wide as well. I can mess around and shift the "functional" lines, but the width of the gaps remain (See the hyperion test video in the background for comparison).

When I touch the 330 ohm resistor connected to the data line, more LEDs start turning on and they all start spasming out. I have tried reinstalling everything but the same problem still prevailed.

I have tried looking everywhere for a solution but I have yet to find someone with a similar issue. I followed Justin Nelson's guide so I didn't install hyperbian but instead use the configuration on the website.

I should probably also mention that there is something up with the LUT? I seem to not have a custom LUT and I don't know which one to install. I got mine from AliExpress and it works well but I don't know what "kind" it is.

Any help is appreciated, thanks in advance! :)

**Comment by awawa-dev** [comment]
Hi
Try to test individual LEDs using 'identify' option from context menu. Check if they react correctly and their positions correlate with the physical location
https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#manual-led-layout-and-testing

**Reply by RSeyda** [reply]
I am able to identify each LED (including the ones that weren't lighting up) but they are all in weird positions. For example, identifying a top middle LED makes a bottom right-hand side LED turn on

**Reply by RSeyda** [reply]
When I check "Reverse Direction" this weird shift disappears but the LEDs light up in a mirrored manner when I identify them

**Reply by awawa-dev** [reply]
Probably geometry is incorrectly designed. Suggest you to start from the beginning. After each step use "identify" function to make sure you are heading in the correct direction. "Reverse Direction" should not be guessed but used only if you designed the LED strip in opposite direction.

**Reply by RSeyda** [reply]
From the perspective of standing in front of the TV screen, It goes clockwise. Is that the "default" direction that most people pick?

**Reply by RSeyda** [reply]
Could you please elaborate on what you mean by start from the beginning and what each step is? I am interpreting it as manually changing the positions of each LED pixel?
I also forgot to mention that I am using the ws281x controller type if that is of any help

**Reply by awawa-dev** [reply]
Reset geometry to default. Set dimension bottom/top/left/right, start to increase input position you should notice how the designer is changing. Use identify to test it all the time.

**Reply by RSeyda** [reply]
I was able to successfully fix the correct input position, but around 4 LEDs on each side of the bottom and top rows as well as the entire left and right LED strips did not light up. I have been trying to experiment by playing around with the number of LEDs on each row and it is somewhat possible to at least get either the top or the bottom row completely lit up and the row parallell to it to be almost fully lit with some weird unfixable shift positions (ex. 2 or 4 LEDs on the side and around the same amount of unlit LEDs on the wanted horizontal row).

**Reply by awawa-dev** [reply]
There is nothing to experiment: fur a such simply setup as on the photo you need to provide dimensions and input position assuming that it's clockwise direction.  If 'identify' for these LEDs wont work then I cant help, probably it's a hardware problem. Maybe these corners connectors are causing problems, who knows.

**Reply by RSeyda** [reply]
The thing is that identify does work on those LEDs. I am able to identify each LED and have them blink without a problem. I have some more pictures here if they might help, I am not that good at describing things haha
![ambilight](https://github.com/user-attachments/assets/25c1984a-06ba-49c0-afc6-69dc75f4cf29)
![image](https://github.com/user-attachments/assets/c42891c8-ce27-4da8-b2df-2cc83273469c)

**Reply by awawa-dev** [reply]
> The thing is that identify does work on those LEDs. I am able to identify each LED and have them blink without a problem. 

then everything is fine for me, I dont have more ideas. The rest depends on the video source.

**Reply by RSeyda** [reply]
I was finally able to solve my problem! You were right about the video source, I was using my laptop which has an aspect ratio of 16:10, which the LED layout didn't account for. By tweaking the advanced settings I was able to make it so that the LEDs would recognize the colors in my video. 
![image](https://github.com/user-attachments/assets/035c0c78-da22-477a-8fc8-e19946e07150)

Thank you so much for your help, and I hope you have a good rest of your day! :D
